cellpack/autopack/DBRecipeHandler.py

```diff
@@ -376,6 +378,37 @@ def should_write(self, db, grad_name):
         return None, None
 
 
+class ResultDoc:
+    def __init__(self, db):
+        self.db = db
+
+    def handle_expired_results(self):
+        """
+        Check if the results in the database are expired and delete them if the linked object expired.
+        """
+        current_utc = datetime.now(timezone.utc)
+        results = self.db.get_all_docs("results")
+        if results:
+            for result in results:
+                result_data = self.db.doc_to_dict(result)
+                result_age = current_utc - result_data["timestamp"]
+                if result_age.days > 180 and not self.validate_existence(
+                    result_data["url"]
+                ):
+                    self.db.delete_doc("results", self.db.doc_id(result))
+            print("Results cleanup complete.")
+        else:
+            print("No results found in the database.")
+
+    def validate_existence(self, url):
+        """
+        Validate the existence of an S3 object by checking if the URL is accessible.
+        Returns True if the URL is accessible.
+        """
+        response = requests.head(url)
+        return response.status_code == 200
```

Comment:
[small nit]: perhaps replace the `200` with `requests.codes.ok`, just in case someone is unfamiliar with http response codes?